### ISSUE TITLE
"param does not exists" error on function destructured with default value

### ISSUE THREAD

**Issue by stanf0rd** [gajus/eslint-plugin-jsdoc#635]
## Expected behavior
No error occured

## Actual behavior
`@param "props.data.className" does not exist on props: eslint(jsdoc/check-param-names)`

## ESLint Config

```js
"jsdoc/check-param-names": "error"
```

## ESLint sample

```js
/**
 * Item
 *
 * @param {object} props
 * @param {object} props.data - case data
 * @param {string} props.data.className - additional css class
 */
export default function Item({
  data: {
    className,
  } = {}, // here! No error occured, if no default value (= {})
}) {
  return ...
```

## Environment

- Node version: 12 LTS
- ESLint version: 7.8.1
- `eslint-plugin-jsdoc` version: 30.4.0

**Comment by stanf0rd** [comment]
So fast! Cool, thank you!
